lms/djangoapps/instructor_task/tasks_helper/components.py

```diff
@@ -0,0 +1,88 @@
+"""
+Instructor tasks related to components and xblocks in the course.
+"""
+
+from datetime import datetime
+from time import time
+from pytz import UTC
+from lms.djangoapps.instructor_analytics.csvs import format_dictlist
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from xmodule.modulestore.django import modulestore
+
+from .runner import TaskProgress
+from .utils import upload_csv_to_report_store
+
+
+def upload_xblock_list_csv(
+    _xblock_instance_args, _entry_id, course_id, task_input, action_name
+):
+    """
+    Generate a csv containing all components.
+    """
+    start_time = time()
+    start_date = datetime.now(UTC)
+
+    overviews = CourseOverview.objects.filter(id=course_id).order_by("id")
+    # NOTE: if we want to report on all courses at once, use this line below instead of the one above
+    # overviews = CourseOverview.objects.all().order_by('id')
+
+    total_courses = overviews.count()
+    task_progress = TaskProgress(action_name, total_courses, start_time)
+
+    current_step = {"step": "Calculating XBlock Info"}
+    task_progress.update_task_state(extra_meta=current_step)
+
+    data = []
+    succeeded_count = 0
+    for overview in overviews:
+        try:
+            course = modulestore().get_course(overview.id)
+            data.extend(
+                {
+                    "Course ID": course.id,
+                    "Course Name": course.display_name,
+                    "Section Name": section.display_name,
+                    "Subsection Name": subsection.display_name,
+                    "Unit Name": unit.display_name,
+                    "Component Name": component.display_name,
+                    "Xblock Type": component.location.block_type,
```

Comment:
This would just return all the component types in a course right? I wonder if it would be better to filter out the default ones (like html, problem, video) from external ones (like pdf, timeline, short answer, etc)?

Reply:
Ah maybe, not sure. We can probably ask the client - they may be interested in all components, for stats like which percentage of the components are third party xblocks.